hack/make/.detect-daemon-osarch

```diff
@@ -16,3 +17,18 @@ export DOCKER_CLIENT_OSARCH="$(docker version | awk '
 	$1 == "Server:" { client = 0; next }
 	client && $1 == "OS/Arch:" { print $2 }
 ')"
+
+# Retrieve the architecture used in contrib/builder/(deb|rpm)/$PACKAGE_ARCH/
+PACKAGE_ARCH="amd64"
+case "$DOCKER_ENGINE_OSARCH" in
+	linux/arm)
+		PACKAGE_ARCH='armhf'
```

Comment:
Added a variable `$PACKAGE_ARCH` for the `contrib/builder/(deb|rpm)/$PACKAGE_ARCH/` sub directories as we want eg. `armhf` instead of `$DOCKER_ENGINE_GOARCH` which is `arm`.